controllers/suite_test.go

```diff
@@ -77,6 +79,17 @@ const (
 )
 
 func TestAPIs(t *testing.T) {
+
+	// check for which deployment type we need to run the tests for
+	envVar := "MODE"
+	testDeployment = os.Getenv(envVar)
+	switch testDeployment {
+	case convergedDeploymentType, providerDeploymentType, consumerDeploymentType:
+	default:
+		panic(fmt.Sprintf("Environment var %s should be set to one of %s, %s and %s values",
+			envVar, convergedDeploymentType, providerDeploymentType, consumerDeploymentType))
+	}
```

Comment:
You have one condition and one test here so why use `switch` instead of an `if`?
```suggestion
	if testDeployment != convergedDeploymentType &&
		testDeployment != providerDeploymentType &&
		testDeployment != consumerDeploymentType {
		panic(fmt.Sprintf("Environment var %s should be set to one of %s, %s and %s values",
			envVar, convergedDeploymentType, providerDeploymentType, consumerDeploymentType))
	}
```

Reply:
- I usually prefer `switch` for more than two args with same result 
- if you disagree, I'll change it to `if` condition

Reply:
- i can validate all required values in a single `case` statement and `default` will be panic

Reply:
I just saw you how to validate using a single if statement. I will argue that because `switch` in go has the opposite semantics from any other C-based language (break is explicit and not fallthrough which is the opposite of any other C-based lang that I am aware of ) then the above is very very misleading for the casual reader as we are so used to have it the other way around